### PR TITLE
Add `detach()` to `HashTextEmbedder`

### PR DESCRIPTION
**Pull request by weihua916**

### CODE REVIEW

**Review by zechengz — APPROVED** [review]

File: torch_frame/testing/text_embedder.py [hunk 1]
```diff
@@ -29,4 +29,4 @@ def __call__(self, sentences: List[str]) -> Tensor:
         idx = torch.LongTensor(
             [hash(s) % self.num_hash_bins for s in sentences],
             device=self.device)
-        return self.embedding(idx)
+        return self.embedding(idx).detach()
```
Comment (rusty1s) [inline torch_frame/testing/text_embedder.py:32]: Sorry for jumping in late. Can you clarify this change for me?

Comment (weihua916) [reply torch_frame/testing/text_embedder.py:32]: Otherwise, `rtb` training will give us [this error](https://discuss.pytorch.org/t/runtimeerror-trying-to-backward-through-the-graph-a-second-time-but-the-buffers-have-already-been-freed-specify-retain-graph-true-when-calling-backward-the-first-time/6795) in the second training iteration.

Comment (rusty1s) [reply torch_frame/testing/text_embedder.py:32]: Oh, this embedder here is not part of the network, right? Then this makes sense. Thanks!

Comment (weihua916) [reply torch_frame/testing/text_embedder.py:32]: That's correct. It's pre-encoding of texts.
